src/utils.js

```diff
@@ -144,3 +144,48 @@ export const doughnutData = {
     },
   ],
 }
+
+//NOTE(Rejon): I didn't want to import lodash just for the merge method. This is a vanilla js implementation.
```

Comment:
For my own education, what are we actually doing here? Is it this:

```Merge algorithms are a family of algorithms that take multiple sorted lists as input and produce a single list as output, containing all the elements of the inputs lists in sorted order.```

And if so, what are we actually merging? 😇

Reply:
Yep, this util method is the bread and butter of how the merging in theme.js works. It's currently imported as "extend", but this algorithm is what allows us to perform a deep merge between objects. Considering the case where darkTheme and defaultTheme share key names this means we can overlap the darkTheme keys on the defaultTheme keys and keep any keys between the two themes that don't exist. The complication comes from the fact that the darkTheme is inside of the defaultTheme object so it can be accessed by keyname. 

This is handed back by default. 
```
theme: {
colors: {
  primary: "Blue",
  secondary: "Green", 
  third: "Yellow",
  
}, 
dark: {
     colors: {
       primary: "Red",
       secondary: "Orange"
     }
   }
}
```

If we scope it down to "dark" for our theme, the extend method returns:
```
theme: {
 colors: {
   primary: "Red",
   secondary: "Orange",
   third: "Yellow"
 }
}
```

But it's also important to notice in my theme default method that I'm using Object.assign since I don't want to directly edit theme.js as I use it. This means you can also just import the entire theme as you need it anywhere in the application. 

While all of this is over-engineered, I believe that it offers the right kind of flexibility to grow without the need to be concerned on it's functionality as the project grows.